apps/balancer-tools/src/app/poolsimulator/(components)/PoolParamsForm.tsx

```diff
@@ -55,110 +58,155 @@ const inputMapper: InputMapperType = {
+    resolver: zodResolver(schemaMapper[data.poolType]),
+    mode: "onChange",
   });
   const {
     register,
     setValue,
-    getValues,
     clearErrors,
+    watch,
     formState: { errors },
   } = form;
 
-  const onSubmit = (data: FieldValues) => {
-    setIsGraphLoading(true);
+  const getOnSubmit = (tabClicked: boolean) => (fieldData: FieldValues) => {
+    const hasNullData = inputMapper[data.poolType].reduce(
+      (sum, input) => sum || !fieldData[input.name],
+      false
+    );
+    if (Object.keys(errors).length || hasNullData) return;
     const dataWithPoolType = {
       poolParams: Object.fromEntries(
-        inputMapper[poolType].map((input) => [input.name, data[input.name]])
+        inputMapper[data.poolType].map((input) => [
+          input.name,
+          input.transformFromFormToData(fieldData[input.name]),
+        ])
       ),
-      tokens: initialData.tokens,
-      poolType,
+      tokens: fieldData.tokens,
+      poolType: data.poolType,
     };
-
-    setInitialData(dataWithPoolType as AnalysisData);
-    setCustomData(dataWithPoolType as AnalysisData);
-    push("/poolsimulator/analysis");
+    setData(dataWithPoolType as AnalysisData);
+    extraOnSubmit?.(dataWithPoolType as AnalysisData, tabClicked);
   };
 
+  function checkDataIsEqualForm() {
+    const fieldData = watch();
+    const dataTransformed = {
+      poolParams: Object.fromEntries(
+        inputMapper[data.poolType].map((input) => [
+          input.name,
+          input.transformFromDataToForm(fieldData[input.name]),
+        ])
+      ),
+      tokens: fieldData.tokens,
+      poolType: data.poolType,
+    };
+    return JSON.stringify(dataTransformed) == JSON.stringify(data);
+  }
+
   useEffect(() => {
     clearErrors();
-    if (initialData == getValues() || !poolType) return;
-    inputMapper[poolType].forEach((input) => {
-      const dataValue = initialData.poolParams?.[input.name];
+    if (checkDataIsEqualForm() || !data.poolType) return;
+    inputMapper[data.poolType].forEach((input) => {
+      const dataValue = data.poolParams?.[input.name];
       if (dataValue) {
-        setValue(input.name, dataValue);
+        setValue(input.name, input.transformFromDataToForm(dataValue));
       }
     });
-    if (initialData?.tokens) setValue("tokens", initialData?.tokens);
-  }, [initialData]);
+    if (data?.tokens) setValue("tokens", data?.tokens);
+  }, [data.poolParams, data.tokens]);
 
   useEffect(() => {
-    clearErrors();
-    Object.entries(inputMapper).forEach(([, value]) => {
-      value.forEach((input) => {
-        setValue(input.name, undefined);
+    function resetForm() {
+      clearErrors();
+      Object.entries(inputMapper).forEach(([, value]) => {
+        value.forEach((input) => {
+          setValue(input.name, undefined);
+        });
       });
-    });
-    setInitialData({
-      poolParams: undefined,
-      tokens: [],
-      poolType: poolType,
-    });
-  }, [poolType]);
+    }
```

Comment:
there's a `reset` method in react-hook-form to clear values/errors. Is that what you're trying to achieve here?

Reply:
it's because we want to reset the inputs to `undefined` to show the placeholder for empty data. But `reset()` expects default data, and with `undefined` the form holds to the last defined value. 

If our default data was like swapFee: 0, then `reset()` works but it won't show the input placeholder